packages/website/src/components/config/utils.ts

```diff
@@ -10,48 +10,65 @@ export interface OptionDeclarations {
   description?: { message: string };
 }
 
-export function parseESLintRC(code?: string): RulesRecord {
+export function parseESLintRC(code?: string): EslintRC {
   if (code) {
     try {
       const parsed: unknown = parse(code);
-      if (isRecord(parsed) && 'rules' in parsed && isRecord(parsed.rules)) {
-        return parsed.rules as RulesRecord;
+      if (isRecord(parsed)) {
+        if ('rules' in parsed && isRecord(parsed.rules)) {
+          return parsed as EslintRC;
+        }
+        return { ...parsed, rules: {} };
       }
     } catch (e) {
       // eslint-disable-next-line no-console
       console.error(e);
     }
   }
-  return {};
+  return { rules: {} };
 }
 
-export function parseTSConfig(code?: string): CompilerFlags {
+export function parseTSConfig(code?: string): TSConfig {
   if (code) {
     try {
       const parsed: unknown = parse(code);
-      if (
-        isRecord(parsed) &&
-        'compilerOptions' in parsed &&
-        isRecord(parsed.compilerOptions)
-      ) {
-        return parsed.compilerOptions as CompilerFlags;
+      if (isRecord(parsed)) {
+        if ('compilerOptions' in parsed && isRecord(parsed.compilerOptions)) {
+          return parsed as TSConfig;
+        }
+        return { ...parsed, compilerOptions: {} };
       }
     } catch (e) {
       // eslint-disable-next-line no-console
       console.error(e);
     }
   }
-  return {};
+  return { compilerOptions: {} };
+}
+
+const moduleRegexp = /^\s*(module\.exports\s*=)?\s*(\{[\s\S]*})\s*;?\s*$/g;
+
+export function tryParseEslintModule(value: string): string {
+  try {
+    if (moduleRegexp.test(value)) {
+      const newValue = toJson(parse(value.replace(moduleRegexp, '$2')));
```

Comment:
as an alternative - you could probably also just `eval` the code if it doesn't start with a `{`.

something like
```ts
function constrainedScopeEval(config) {
    const module = {exports: {}};
    eval(config);
    return module.exports;
}
```

Reply:
initially I did something similar, but with scope isolation, but there are to many edge cases, we have to filter imports etc,

and I decided against it

Reply:
could we just not filter any of the garbage and let it go?
I think we could just wrap it in a try/catch and then show any thrown errors either as a monaco overlay or by some other means?

If the user dumps junk and it breaks - do we care?

The reason I like the eval route is that it lets people use variables and functions if they want.
But we can punt on that if you think it's not worth the effort!

Reply:
ok, i'm going to change to to evaluation, but instead of eval witch is supper bad for browser optimization, I'm going to opt in for `new Function` as it doesn't force browser to disable optimization and provides "basic" scope isolation